src/contracts/core/DelegationManager.sol

```diff
@@ -103,9 +107,11 @@ contract DelegationManager is
      */
     function registerAsOperator(
         OperatorDetails calldata registeringOperatorDetails,
+        uint32 allocationDelay,
         string calldata metadataURI
     ) external {
```

Comment:
If we are comfortable changing this interface, let's make it a little more sane and use a new version of the `OperatorDetails` struct that  includes an `allocationDelay` and does not have deprecated fields

Reply:
`OperatorDetails` is only storing `delegationApprover` at this point, I'd say just remove this OperatorDetails struct from the interface and only use it to store for backwards compatability.
Perhaps something like the following
```
function registerAsOperator(
        address delegationApprover,
        uint32 allocationDelay,
        string calldata metadataURI
    ) external {
```